soc/silabs/silabs_siwx91x/siwg917/nwp.c

```diff
@@ -87,10 +113,76 @@ static int siwg917_nwp_init(void)
 					SL_SI91X_BLE_AE_MAX_ADV_SETS(RSI_BLE_AE_MAX_ADV_SETS);
 #endif
 
+	if (wifi_oper_mode == SL_SI91X_CLIENT_MODE) {
+		memcpy(get_config, &default_nwp_config, sizeof(default_nwp_config));
+		return 0;
+	} else if (wifi_oper_mode == SL_SI91X_ACCESS_POINT_MODE) {
+		memcpy(get_config, &default_ap_configuration, sizeof(default_ap_configuration));
+		return 0;
+	}
+
+	return -EINVAL;
+}
+
+int siwx91x_nwp_mode_switch(uint8_t oper_mode)
+{
+	sl_wifi_device_configuration_t nwp_config;
+	sl_wifi_interface_t interface;
+	sl_status_t status;
+
+	interface = sl_wifi_get_default_interface();
+	switch (oper_mode) {
+	case SL_SI91X_CLIENT_MODE:
+		if (FIELD_GET(SIWX91X_INTERFACE_MASK, interface) == SL_WIFI_CLIENT_INTERFACE) {
+			return 0;
+		}
+
+		siwg91x_get_nwp_config(SL_SI91X_CLIENT_MODE, &nwp_config);
+		break;
+	case SL_SI91X_ACCESS_POINT_MODE:
+		if (FIELD_GET(SIWX91X_INTERFACE_MASK, interface) == SL_WIFI_AP_INTERFACE) {
+			return 0;
+		}
+
+		siwg91x_get_nwp_config(SL_SI91X_ACCESS_POINT_MODE, &nwp_config);
+		break;
+	default:
+		return -EINVAL;
+	}
```

Comment:
I don't have the full picture, but maybe `siwx91x_nwp_mode_switch()` could take a `WIFI_AP_MODE` / `WIFI_STA_MODE` as parameter. So, we could keep the `oper_mode` stuff internal to `nwp.c` and we won't have to expose it to the wifi driver.